utf8mb4-conversion.php

```diff
@@ -118,7 +118,7 @@
         $name = $row[0];
         $type = $row[1];
         $allownull = (strtoupper($row[2]) == 'YES') ? 'NULL' : 'NOT NULL';
-        $defaultval = (trim($row[4]) == '') ? '' : "DEFAULT '{$row[4]}'";
+        $defaultval = (empty($row[4])) ? "DEFAULT ' '" : "DEFAULT '{$row[4]}'";
```

Comment:
Two things: 

We can't use `empty()` here because one might have intentionally set the default on an `INT` field to `0`, which `empty()` will misinterpret.

`DEFAULT ' '` as a "single space" isn't ideal; and we don't want to "always set a default", which is what this change would do.

Reply:
Perhaps better is to:
- check `$allownull` (since, at least in Zen Cart use, we don't set a DEFAULT if NULL is allowed (granted, I'm saying that off the top of my head, without inspecting the entire schema))
- maybe check whether $row[4] (which is the `Default` value according to `DESCRIBE table`) to see whether it is null. I haven't inspected deeply: is this value `null` if no default is set? If so, then perhaps we can use `is_null()` instead of the original `trim() == ''`?